lib/insert/insert_refunds.rb

```diff
@@ -66,8 +66,28 @@ def self.with_stripe(charge, h)
     refund_row = Qx.update(:refunds).set(payment_id: payment_row['id']).ts.where(id: refund_row['id']).returning('*').execute.first
     # Update original payment to increment its refund_total for any future refund attempts
     Qx.update(:payments).set("refund_total=refund_total + #{h['amount'].to_i}").ts.where(id: original_payment['id']).execute
+
+    refund = Refund.find(refund_row['id'])
+    refund_payment = Payment.find(payment_row['id'])
+    refund_payment.update(refund: refund)
+
+    stripe_transaction_charge = StripeCharge.find_by(payment: original_payment['id'])
+    stripe_transaction_refund = StripeRefund.new(payment: refund_payment)
+    transaction_id = stripe_transaction_charge.subtransaction_payment.subtransaction.transaction_id
+    transaction = Transaction.find(transaction_id)
+
+    transaction.amount += gross
+    refund_subtransaction_payment = transaction.subtransaction.subtransaction_payments.create(paymentable: stripe_transaction_refund)
+
+    stripe_transaction_refund.save!
+    refund_subtransaction_payment.save!
+    transaction.save!
+
+    transaction.publish_updated
+    transaction.publish_refunded
+    refund_subtransaction_payment.publish_created
+
     # Send the refund receipts in a delayed job
-    Houdini.event_publisher.announce(:create_refund, Refund.find(refund_row['id']))
```

Comment:
It seemed outdated but I'm not sure, could it really be removed?

Reply:
Let's keep it for now.

Reply:
Okay, I'll add a test for it too